src/components/TasksTable/index.tsx

```diff
@@ -172,9 +175,24 @@ const getColumnDefs = ({
       multiple: 4,
     },
     className: "cy-task-table-col-STATUS",
-    render: (status: string, { execution, id }) =>
-      status && (
-        <TaskStatusBadge status={status} id={id} execution={execution} />
+    render: (status: string, { dependsOn, execution, id }) =>
+      dependsOn?.length && status === TaskStatus.Blocked ? (
+        <Tooltip
+          data-cy="depends-on-tooltip"
+          justify="middle"
+          trigger={
+            <span>
+              <TaskStatusBadge status={status} id={id} execution={execution} />
+            </span>
+          }
+        >
+          Depends on <>{pluralize("task", dependsOn.length)}</>:{" "}
```

Comment:
Is the fragment necessary?
```suggestion
          Depends on {pluralize("task", dependsOn.length)}:{" "}
```